integration_tests/tests/billing-and-transaction-view-tests.ts

```diff
@@ -0,0 +1,148 @@
+    ${"No BillingQty No Billing Price (no invoice) but has TransactionQty TransactionPrice"} | ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client1"} | ${TimeStamps.CURRENT_TIME} | ${"1"}             | ${"-3.3300"} | ${"-1"} | ${"-100.0000"}         | ${"-100"}            | ${undefined} | ${undefined} | ${"3.3300"}      | ${"1"}
+  `(
+    "results retrived from billing and transaction_curated view query should match with expected $testCase,$billingQuantity,$priceDiff,$qtyDiff,$priceDifferencePercent,$qtyDifferencePercent,$billingPrice",
+    async (data) => {
+      await assertResultsWithTestData(data);
+    }
+  );
+});
+
+
+interface TestData{
+  eventName: EventName,
+  clientId: ClientId,
+  eventTime: TimeStamps,
+  numberOfTestEvents: number,
+  priceDiff: string,
+  qtyDiff: string,
+  priceDifferencePercent: string,
+  qtyDifferencePercent: string,
+  billingPrice: string,
+  billingQty: string,
+  transactionPrice: string,
+  transactionQty: string
+}
+
+export const assertResultsWithTestData = async ({
+  eventName,
+  clientId,
+  eventTime,
+  numberOfTestEvents,
+  priceDiff,
+  qtyDiff,
+  priceDifferencePercent,
+  qtyDifferencePercent,
+  billingPrice,
+  billingQty,
+  transactionPrice,
+  transactionQty}:TestData
+): Promise<void> => {
+  const eventIds = await generatePublishAndValidateEvents({
+    numberOfTestEvents,
+    eventName,
+    clientId,
+    eventTime,
+  });
+  const tableName = TableNames.BILLING_TRANSACTION_CURATED;
+  const year = new Date(eventTimeStamp[eventTime] * 1000).getFullYear();
+  const response: BillingTransactionCurated[] =
+    await queryResponseFilterByVendorServiceNameYear({
+      clientId,
+      eventName,
+      tableName,
+      year,
+    });
+  await deleteS3Events(eventIds, eventTime);
+  expect(response[0].price_difference).toEqual(priceDiff);
+  expect(response[0].quantity_difference).toEqual(qtyDiff);
+  expect(response[0].price_difference_percentage).toEqual(
+    priceDifferencePercent
+  );
+  expect(response[0].quantity_difference_percentage).toEqual(
+    qtyDifferencePercent
+  );
+  expect(response[0].billing_price).toEqual(billingPrice);
+  expect(response[0].billing_quantity).toEqual(billingQty);
+  expect(response[0].transaction_price).toEqual(transactionPrice);
+  expect(response[0].transaction_quantity).toEqual(transactionQty);
+};
+
+interface BillingTransactionCurated {
+  vendor_name: string;
+  service_name: string;
+  year: string;
+  month: string;
+  price_difference: number;
+  quantity_difference: number;
+  price_difference_percentage: number;
+  quantity_difference_percentage: number;
+  billing_price: number;
+  billing_quantity: number;
+  transaction_price: number;
+  transaction_quantity: number;
+}
```

Comment:
You could make this

```
type BillingTransactionCurated = Array<{
  vendor_name: string;
  service_name: string;
  year: string;
  month: string;
  price_difference: number;
  quantity_difference: number;
  price_difference_percentage: number;
  quantity_difference_percentage: number;
  billing_price: number;
  billing_quantity: number;
  transaction_price: number;
  transaction_quantity: number;
}>
```

then you won't need the [] on line 113

Reply:
fixed